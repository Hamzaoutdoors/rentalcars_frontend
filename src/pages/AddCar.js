import styled from 'styled-components';
import CarForm from '../components/cars/CarForm';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem;
  min-heigth: 100vh;
  width: 100%;
`;

const Title = styled.div`
  display: flex;
  justify-content: center;
  border-bottom: 2px solid black;
  margin-bottom: 1rem;
  font-size: 36px;
  font-weight: 700;
  font-family: 'Bebas Neue', 'Arial', cursive !important;
  color: black;
`;

const SubTitle = styled.div`
  font-size: 20px;
  font-weight: 500;
  padding: 0 2.5rem;
  font-family: 'Bebas Neue', 'Arial', cursive !important;
  color: black;
`;
const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 2rem;
`;

const AddCar = () => (
  <Container>
    <Container>
      <Wrapper>
        <Title>
          <p style={{ padding: '0 2.5rem', margin: 0 }}>
            Add a Car to rent
          </p>
        </Title>
        <SubTitle>
          Let&apos;s be partners, with a 2% fee you can annouce your car on our website!
          We are available in 10 cities!
        </SubTitle>
      </Wrapper>
      <CarForm />
    </Container>
  </Container>
);

export default AddCar;
