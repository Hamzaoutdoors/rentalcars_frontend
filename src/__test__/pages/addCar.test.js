import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import AddCar from '../../pages/AddCar';
import store from '../../redux/configureStore';

it('render the Add Car page correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <AddCar />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
