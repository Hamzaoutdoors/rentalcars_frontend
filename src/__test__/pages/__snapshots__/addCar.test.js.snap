// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the Add Car page correctly 1`] = `
<div
  className="sc-iCfMLu dfaXPl"
>
  <div
    className="sc-jrQzAO jHbGCH"
  >
    <div
      className="sc-furwcr dgcOZg"
    >
      <p
        style={
          Object {
            "margin": 0,
            "padding": "0 2.5rem",
          }
        }
      >
        Add a Car to rent
      </p>
    </div>
    <div
      className="sc-pVTFL jvwajp"
    >
      Let's be partners, with a 2% fee you can annouce your car on our website! We are available in 10 cities!
    </div>
  </div>
  <form
    className="sc-bdvvtL ljALzO"
    onSubmit={[Function]}
  >
    <div
      className="sc-gsDKAQ cHUBvv"
    >
      <div
        className="sc-dkPtRN hzTcze"
      >
        <div
          className="sc-hKwDye dUcxzm"
        >
          Car Info
        </div>
        <input
          className="sc-jRQBWg bQZufB"
          defaultValue=""
          id="car-model"
          name="name"
          placeholder="Model"
          type="text"
        />
        <input
          className="sc-jRQBWg bQZufB"
          defaultValue=""
          id="car-brand"
          name="brand"
          placeholder="Brand"
          type="text"
        />
        <input
          className="sc-jRQBWg bQZufB"
          defaultValue=""
          id="car-image"
          name="imgUrl"
          placeholder="Image Url"
          type="text"
        />
      </div>
    </div>
    <div
      className="sc-gsDKAQ cHUBvv"
    >
      <div
        className="sc-dkPtRN hzTcze"
      >
        <div
          className="sc-hKwDye dUcxzm"
        >
          Car Details
        </div>
        <input
          className="sc-jRQBWg bQZufB"
          defaultValue=""
          id="price-daily"
          name="price_daily"
          placeholder="Price Daily"
          step="0.01"
          type="number"
        />
        <input
          className="sc-jRQBWg bQZufB"
          defaultValue=""
          id="price-monthly"
          name="price_monthly"
          placeholder="Price Monthly"
          step="0.01"
          type="number"
        />
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <div
            className="sc-eCImPb jFUiIc"
          >
            Color:
          </div>
          <input
            className="sc-jRQBWg bQZufB"
            defaultValue=""
            id="car-color"
            name="color"
            placeholder="Main color (hex value)"
            style={
              Object {
                "height": "35px",
                "width": "100px",
              }
            }
            type="color"
          />
        </div>
      </div>
      <div
        className="sc-dkPtRN hzTcze"
      >
        <button
          className="sc-gKclnd KxBTB"
          type="submit"
        >
          Announce
        </button>
      </div>
    </div>
  </form>
</div>
`;
